src/nvim/highlight_group.c

```diff
@@ -301,6 +301,11 @@ static const char *highlight_init_both[] = {
   "default link @tag         Tag",
   "default link @tag.builtin Special",
 
+  // Higlight "===" and "---" heading delimiters specially.
+  // This is for :help but also presumably useful for markdown etc.
+  "default @markup.heading.1.delimiter guibg=bg guifg=bg guisp=fg gui=underdouble,nocombine ctermbg=NONE ctermfg=NONE cterm=underdouble,nocombine",
```

Comment:
```suggestion
  "default @markup.heading.1.delimiter.vimdoc guibg=bg guifg=bg guisp=fg gui=underdouble,nocombine ctermbg=NONE ctermfg=NONE cterm=underdouble,nocombine",
```

Reply:
Now that it's specific to vimdoc, we should be targeted in the group definition.